Tests/UnitTests/AdyenTheme/AdyenThemeTests.swift

```diff
@@ -95,5 +95,174 @@ final class AdyenThemeTests: XCTestCase {
+        newTextFieldStyle.backgroundColor = AdyenColorScheme.default.primary
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.backgroundColor, AdyenColorScheme.default.primary)
+    }
+    
+    func test_textfieldMethod_shouldUpdateTextColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.textColor = AdyenColorScheme.default.textOnPrimary
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.textColor, AdyenColorScheme.default.textOnPrimary)
+    }
+    
+    func test_textfieldMethod_shouldUpdateActiveColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.activeColor = AdyenColorScheme.default.highlight
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.activeColor, AdyenColorScheme.default.highlight)
+    }
+    
+    func test_textfieldMethod_shouldUpdateErrorColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.errorColor = AdyenColorScheme.default.textOnDestructive
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.errorColor, AdyenColorScheme.default.textOnDestructive)
+    }
+    
+    func test_textfieldMethod_shouldUpdateBorderColor() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        newTextFieldStyle.borderColor = AdyenColorScheme.default.outline
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.borderColor, AdyenColorScheme.default.outline)
+    }
+    
+    func test_textFieldMethod_shouldUpdateCornerRadius() {
+        // Given
+        let cornerRadius = CornerRounding.fixed(10.0)
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.cornerRadius = cornerRadius
+        
+        // When
+        let updatedTheme = theme.textfield(newTextFieldStyle)
+        
+        // Then
+        XCTAssertEqual(updatedTheme.textFieldStyle.cornerRadius, cornerRadius)
+    }
+    
+    func test_textFieldMethod_shouldUpdateBorderWidth() {
+        // Given
+        let theme = AdyenTheme()
+        
+        var newTextFieldStyle = AdyenTextFieldStyle()
+        
+        newTextFieldStyle.borderWidth = 2.0
```

Comment:
Using magic numbers makes code less readable. It's a good practice to use following or similar:

```swift
let expectedBorderWidth: CGFloat = 2.0
newTextFieldStyle.borderWidth = expectedBorderWidth
// the rest
XCTAssertEqual(updatedTheme.textFieldStyle.borderWidth, expectedBorderWidth)
```

The same applies to the rest of the test cases.